core/src/main/java/io/questdb/cutlass/http/processors/TextLoaderCompletedState.java

```diff
@@ -38,11 +37,9 @@ public void copyState(TextLoader textLoader) {
         // Some values are come from TableWriter and has to be copied
         // in order to release TableWriter back to the Engine
         this.writtenLineCount = textLoader.getWrittenLineCount();
-        this.metadata = textLoader.getMetadata() != null
-                ? GenericRecordMetadata.copyOf(textLoader.getMetadata())
-                : null;
         // Some values are safe to get from TextLoader
         this.textLoader = textLoader;
+        this.metadata = textLoader.getMetadata();
```

Comment:
Why no copy? The copy is taken to release the writer in case of sending response is slow so that the writer is not blocked for long. When the writer is released it can go back to the pool and the metadata reference here can belong to a closed writer.